packages/core/src/components/ColorPalette/ColorPicker.tsx

```diff
@@ -1,35 +1,165 @@
-import { Box } from '@chakra-ui/react'
-import { Color, ColorResult, SketchPicker } from '@hello-pangea/color-picker'
-
-export function ColorPicker({
-  colorPickerColor,
-  onChange,
-  presetColors,
-}: {
-  presetColors: string[]
-  colorPickerColor: Color
-  onChange: (
-    colorPickerColor: ColorResult,
-    event: React.MouseEvent<Element, MouseEvent> | undefined
-  ) => void
-}) {
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Button,
+  Input,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+} from '@chakra-ui/react'
+import { useState } from 'react'
+import { HexColorPicker, RgbaColorPicker } from 'react-colorful'
+import { AnyColor } from 'react-colorful/dist/types'
+import tinycolor from 'tinycolor2'
+
+const formats = ['HEX', 'RGB'] as const
+
+interface Props {
+  colorPickerColor: AnyColor
+  onChange: ((color: AnyColor) => void)
+}
+
+function DropdownInput({ onChange, colorPickerColor }: Props) {
+  const colorParsed = tinycolor(colorPickerColor);
+  const type = colorParsed.getFormat().toUpperCase();
```

Comment:
Does fetching this type from the color work, instead of saving it from store. Thoughts @ktra99 ?